cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java

```diff
@@ -119,6 +135,9 @@ public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(deviceType.name());
         dest.writeInt(platformId);
         dest.writeInt(productId);
+        dest.writeString(ipAddress);
```

Comment:
Are these nullable fields?  If so, we should be using `writeValue()` and `readValue()`.